libres/lib/res_util/memory.cpp

```diff
@@ -0,0 +1,114 @@
+/*
+   Copyright (C) 2011  Equinor ASA, Norway.
+
+   The file 'memory.cpp' is part of ERT - Ensemble based Reservoir Tool.
+
+   ERT is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   ERT is distributed in the hope that it will be useful, but WITHOUT ANY
+   WARRANTY; without even the implied warranty of MERCHANTABILITY or
+   FITNESS FOR A PARTICULAR PURPOSE.
+
+   See the GNU General Public License at <http://www.gnu.org/licenses/gpl.html>
+   for more details.
+*/
+
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <sys/resource.h>
+
+/**
+ * NOTE: Linux only.
+ *
+ * Returns available memory (Mb) on a Linux system or -1 if something failed.
+ *
+ * For details and explanation of the returned value, see
+ * the field MemFree in the section on /proc/meminfo in
+ *
+ *   https://www.kernel.org/doc/Documentation/filesystems/proc.txt
+ */
+long SystemRamFree(void) {
+    std::ifstream meminfo;
+    meminfo.open("/proc/meminfo", std::ios::in);
+    if (meminfo.is_open()) {
+        std::string tp;
+        int ram;
+        while (std::getline(meminfo, tp)) {
+            if (sscanf(tp.data(), "MemFree: %d kB", &ram) == 1)
+                return ram / 1000;
+        }
+    }
+    return -1;
+}
+
+/**
+ * NOTE: Linux only.
+ *
+ * Returns the current memory (Mb) used by the process,
+ * or -1 if something failed.
+ *
+ * For details and explanation of the returned value, see
+ * the field VmSize in section on /proc/self/status in
+ *
+ *   https://www.kernel.org/doc/Documentation/filesystems/proc.txt
+ */
+long ProcessMemory(void) {
```

Comment:
Looks like three of these functions could be implemented with a common private implementation:
```C++
long parse_mem(const std::string& fname, const std::string& sscanf_fmt) {
     
}
```